library/ssl_tls13_client.c

```diff
@@ -2203,6 +2207,18 @@ static int ssl_tls13_process_encrypted_extensions(mbedtls_ssl_context *ssl)
     }
 #endif
 
+    /*
+     * Move `session_negotiate->ciphersuite` assignment here which after
+     * early data cipher suite check when receiving "early_data" extension
+     * in EncryptedExtensions.
+     *
+     * We compute transform_handshake by the cipher suite chosen from
+     * the server in `handshake`. `session_negotiate->ciphersuite` is the
+     * cipher suite negotiated in previous connection and it is not used for
+     * computing transform_handshake.
```

Comment:
```suggestion
     * the server in `handshake`. `session_negotiate->ciphersuite` is the
     * cipher suite proposed in this session and not used for
     * computing transform_handshake.
```

Reply:
The value is not come from `previous connection` although they are same in most cases

Reply:
That's a concept issue. I mean the values are same but they are not same.

In theory, we can send different ciphersuite in resumption session(not early data), that is not forbid by spec(please correct me.)